cpu/lm4f120/periph/spi.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2017 Marc Poulhiès
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_lm4f120
+ * @{
+ *
+ * @file
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Marc Poulhiès <dkm@kataplop.net>
+ *
+ * @}
+ */
+#include "cpu.h"
+#include "mutex.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+#include "periph_conf.h"
+#include "board.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#if SPI_NUMOF
+
+/**
+ * @brief Array holding one pre-initialized mutex for each SPI device
+ */
+static mutex_t locks[SPI_NUMOF];
+
+struct spi_conf_s {
+    unsigned long ssi_sysctl;
+    unsigned long ssi_base;
+    unsigned long gpio_sysctl;
+    unsigned long gpio_port;
+    struct spi_pin_conf_s {
+        unsigned long clk;
+        unsigned long fss;
+        unsigned long rx;
+        unsigned long tx;
+        unsigned long mask;
+    } pins;
+};
+
+static const struct spi_conf_s spi_confs[SPI_NUMOF] = {
+#if (SPI_NUMOF >= 1)
+    {
+        .ssi_sysctl = SYSCTL_PERIPH_SSI0,
+        .ssi_base = SSI0_BASE,
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOA,
+        .gpio_port = GPIO_PORTA_BASE,
+        .pins = {
+            .clk = GPIO_PA2_SSI0CLK,
+            .fss = GPIO_PA3_SSI0FSS,
+            .rx = GPIO_PA4_SSI0RX,
+            .tx = GPIO_PA5_SSI0TX,
+            .mask = GPIO_PIN_2 | GPIO_PIN_3 | GPIO_PIN_4 | GPIO_PIN_5
+        }
+    },
+#endif
```

Comment:
All these are board specific peripheral configurations right? I would prefer to fine them in the periph_conf.h as well. Compare the [stm32f4discovery](https://github.com/RIOT-OS/RIOT/blob/master/boards/stm32f4discovery/include/periph_conf.h#L208)